File: java/pe/src/ktn/lab/Perf.java
package ktn.lab;

import java.math.BigInteger;

public class Perf {
    static BigInteger fib(int n) {
        BigInteger a = BigInteger.ONE;
        BigInteger b = BigInteger.ZERO;
        BigInteger c;

        for (int i = 1; i < n; ++i) {
            c = a;
            a = a.add(b);
            b = c;
        }

        return a;
    }
}

File: kt/pe/src/main/kotlin/ktn/lab/Perf.kt
package ktn.lab

import java.math.BigInteger

object Perf {
    internal fun fib(n: Int): BigInteger {
        var a = BigInteger.ONE
        var b = BigInteger.ZERO
        var c: BigInteger

        for (i in 1 until n) {
            c = a
            a = a.add(b)
            b = c
        }

        return a
    }
}
